src/v/cluster/cluster_discovery.cc

```diff
@@ -142,8 +202,259 @@ ss::future<bool> cluster_discovery::dispatch_node_uuid_registration_to_seeds(
     co_return false;
 }
 
-bool cluster_discovery::is_cluster_founder() const {
-    return config::node().seed_servers().empty();
+ss::future<cluster_bootstrap_info_reply>
+cluster_discovery::request_cluster_bootstrap_info_single(
+  const net::unresolved_address addr) const {
+    vlog(clusterlog.info, "Requesting cluster bootstrap info from {}", addr);
+    _as.check();
+    for (auto repeat_jitter = simple_time_jitter<model::timeout_clock>(1s);;) {
+        result<cluster_bootstrap_info_reply> reply_result(
+          std::errc::connection_refused);
+        try {
+            reply_result = co_await do_with_client_one_shot<
+              cluster_bootstrap_client_protocol>(
+              addr,
+              config::node().rpc_server_tls(),
+              2s,
+              [](cluster_bootstrap_client_protocol c) {
+                  return c
+                    .cluster_bootstrap_info(
+                      cluster_bootstrap_info_request{},
+                      rpc::client_opts(rpc::clock_type::now() + 2s))
+                    .then(&rpc::get_ctx_data<cluster_bootstrap_info_reply>);
+              });
+            if (reply_result) {
+                vlog(
+                  clusterlog.info,
+                  "Obtained cluster bootstrap info from {}",
+                  addr);
+                vlog(clusterlog.debug, "{}", reply_result.value());
+                co_return std::move(reply_result.value());
+            }
+            vlog(
+              clusterlog.debug,
+              "Cluster bootstrap info failed with {}",
+              reply_result.error());
+        } catch (...) {
+            vlog(
+              clusterlog.warn,
+              "Error requesting cluster bootstrap info from {}, retrying. {}",
+              addr,
+              std::current_exception());
+        }
+        co_await ss::sleep_abortable(repeat_jitter.next_duration(), _as);
+        vlog(clusterlog.trace, "Retrying cluster bootstrap info from {}", addr);
+    }
+}
+
+ss::future<> cluster_discovery::fetch_cluster_bootstrap_info() {
+    if (_cluster_bootstrap_info_fetched) {
+        co_return;
+    }
+    const auto units = co_await _fetch_cluster_bootstrap_mx.get_units(_as);
+    if (_cluster_bootstrap_info_fetched) {
+        co_return;
+    }
```

Comment:
nit: this is surprising to see because we only ever call this in series. I think having this here will give readers the wrong impression that they should expect parallel calls, unless this is something you intend on adding immediately.